src/common/offline_transformations/src/compress_quantize_weigths.cpp

```diff
@@ -858,49 +964,108 @@ static std::shared_ptr<ov::op::v0::Constant> compress_quantized_weights_internal
-
-    auto f = [compressed_weights_with_fused_zero_point,
-              levels_minus_one,
-              output_low,
-              output_high,
-              zero_point_is_zero,
-              convert_to_low_precision,
-              &can_fuse_zero_point](T weights_value, T input_low, T input_high, T zero_point) mutable {
-        int8_t compressed_weights_value = convert_to_low_precision(
-            quantize(weights_value, input_low, input_high, output_low, output_high, levels_minus_one));
-        if (!zero_point_is_zero && can_fuse_zero_point) {
-            T weights_minus_zero_point = static_cast<T>(compressed_weights_value) - zero_point;
-            int8_t compressed_weights_with_fused_zero_point_value = convert_to_low_precision(weights_minus_zero_point);
-            can_fuse_zero_point &=
-                std::fabs(compressed_weights_with_fused_zero_point_value - weights_minus_zero_point) < 1e-4;
-            *compressed_weights_with_fused_zero_point++ = compressed_weights_with_fused_zero_point_value;
+
+    if (tensor_el_type == ov::element::i8) {
+        int8_t* compressed_weights = compressed_weights_tensor.data<int8_t>();
+        int8_t* compressed_weights_with_fused_zero_point = nullptr;
+
+        if (!zero_point_is_zero) {
+            compressed_weights_with_fused_zero_point = compressed_weights_with_fused_zero_point_tensor.data<int8_t>();
         }
-        return compressed_weights_value;
-    };
+        T levels_minus_one = static_cast<T>(levels - 1);
+        T output_low = -static_cast<T>(levels / 2);
+        T output_high = levels_minus_one + output_low;
+        can_fuse_zero_point = !zero_point_is_zero;
+        const auto convert_to_low_precision = low_precision_type == ov::element::i4 ? convert_to_int4 : convert_to_int8;
+
+        auto f = [compressed_weights_with_fused_zero_point,
+                  levels_minus_one,
+                  output_low,
+                  output_high,
+                  zero_point_is_zero,
+                  convert_to_low_precision,
+                  &can_fuse_zero_point](T weights_value, T input_low, T input_high, T zero_point) mutable {
+            int8_t compressed_weights_value = convert_to_low_precision(
+                quantize(weights_value, input_low, input_high, output_low, output_high, levels_minus_one));
+            if (!zero_point_is_zero && can_fuse_zero_point) {
+                T weights_minus_zero_point = static_cast<T>(compressed_weights_value) - zero_point;
+                int8_t compressed_weights_with_fused_zero_point_value =
+                    convert_to_low_precision(weights_minus_zero_point);
+                can_fuse_zero_point &=
+                    std::fabs(compressed_weights_with_fused_zero_point_value - weights_minus_zero_point) < 1e-4;
+                *compressed_weights_with_fused_zero_point++ = compressed_weights_with_fused_zero_point_value;
+            }
```

Comment:
As I understand, if the above check `if (!zero_point_is_zero)` will be false, `compressed_weights_with_fused_zero_point` will remain nullptr and we'll try to increment a nullptr pointer. Or is it an impossible case?

Reply:
we have a check "if (!zero_point_is_zero && can_fuse_zero_point) " on line 1005, so it shouldn't be nullptr here